<launch>
    <!-- parameters -->
    <arg name="dev_name" default="/dev/ttyUSB0" doc="device file of the IMU"/>
    <arg name="baudrate" default="115200" doc="baudrate of the IMU"/>
    <arg name="timeout" default="0.002" doc="timeout for the IMU communication"/>
    <arg name="stream_type" default="serial" doc="how to communication with xsens"/>
    <arg name="frame_id" default="/imu" doc="frame id of the IMU"/>
    <arg name="frame_local" default="ENU" doc="desired frame orientation (ENU, NED or NWU)"/>
    <arg name="no_rotation_duration" default="0" doc="duration (int in seconds) of the no-rotation calibration procedure"/>
    <arg name="ns" default="xsens_driver"/>

    <group>
        <node pkg="device_xsens" type="xsens_node" name="$(arg ns)" output="screen" >
            <param name="ns" value="$(arg ns)" />
            <param name="heartbeat_id" value="xsens"/>
            <param name="dev_name" value="$(arg dev_name)"/>
            <param name="stream_type" value="$(arg stream_type)"/>
            <param name="baudrate" value="$(arg baudrate)"/>
            <param name="timeout" value="$(arg timeout)"/>
            <param name="frame_id" value="$(arg frame_id)"/>
            <param name="frame_local" value="$(arg frame_local)"/>
            <param name="no_rotation_duration" value="$(arg no_rotation_duration)"/>
        </node>
    </group>

</launch>
